### PR TITLE
HalfCauchy, Gamma, Weibull and LogNormal moments

### PR DESCRIPTION
**Pull request by lucianopaz**
Add moments and tests for the below distributions as part of #5078

- pymc.distributions.continuous.HalfCauchy
- pymc.distributions.continuous.Gamma
- pymc.distributions.continuous.Weibull
- pymc.distributions.continuous.LogNormal

### CODE REVIEW

File: pymc/distributions/continuous.py [hunk 1]
```diff
@@ -2100,6 +2106,12 @@ def dist(cls, beta, *args, **kwargs):
         assert_negative_support(beta, "beta", "HalfCauchy")
         return super().dist([0.0, beta], **kwargs)
 
+    def get_moment(rv, size, loc, beta):
+        mean, _ = at.broadcast_arrays(at.as_tensor(0.0, dtype=rv.dtype), beta)
```
Comment (ricardoV94) [inline pymc/distributions/continuous.py]: Does something like this also work?

```suggestion
        mean = at.zeros_like(beta)
```

Comment (ricardoV94) [reply pymc/distributions/continuous.py]: Is zero a good point to start sampling from this distribution?

**Comment by ricardoV94** [comment]
@lucianopaz seems like pre-commit is failing with import sorting

File: pymc/distributions/continuous.py (hunk 1 above)
Comment (ricardoV94) [reply pymc/distributions/continuous.py]: Looking at V3 I think the median (beta) takes precedence over the mode (0), so we were probably using the former to initialize halfcauchys

Comment (lucianopaz) [reply pymc/distributions/continuous.py]: Oh, I'll use the median instead then. Maybe we should list the order of preference for the moments in #5078

**Comment by lucianopaz** [comment]
> @lucianopaz seems like pre-commit is failing with import sorting

Yeah I messed up the pre commit while rebasing the PR. But now everything should be fixed

**Review by twiecki — APPROVED** [review]

**Comment by twiecki** [comment]
👍

**Pull request merged**